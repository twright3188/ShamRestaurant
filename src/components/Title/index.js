import React from 'react';
import PropTypes from 'prop-types';
import { Text, StyleSheet } from 'react-native';
import Colors from '../../theme';

const styles = StyleSheet.create({
  captionText: {
    position: 'relative',
    width: '100%',
    color: Colors.caption,
    alignItems: 'center',
    justifyContent: 'center',
    fontFamily: Colors.FontBaseBold,
    fontWeight: 'bold',
    flexWrap: 'wrap'
  }
});

function Title(props) {
  const { children, fontSize, style } = props;

  return (
    <Text style={[styles.captionText, { fontSize , lineHeight: fontSize * 1.142 }, style]} numberOfLines={1}>{children}</Text>
  );
}

Title.defaultProps = {
  fontSize: 35,
  children: '',
  
  style: {}
};

Title.propTypes = {
  fontSize: PropTypes.number,
  children: PropTypes.node,
  style: PropTypes.shape({ root: PropTypes.string })
};

export default Title;