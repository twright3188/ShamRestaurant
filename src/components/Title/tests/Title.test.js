import React from 'react';
import renderer from 'react-test-renderer';
import Title from '../index';

it('title render', () => {
    const tree = renderer.create(<Title>test</Title>).toJSON();
    expect(tree).toMatchSnapshot();
});