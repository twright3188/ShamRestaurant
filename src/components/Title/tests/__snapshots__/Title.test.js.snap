// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title render 1`] = `
<Text
  style={
    Array [
      Object {
        "alignItems": "center",
        "color": "#222624",
        "flexWrap": "wrap",
        "fontFamily": "Helvetica Neue",
        "fontWeight": "bold",
        "justifyContent": "center",
        "position": "relative",
        "width": "100%",
      },
      Object {
        "fontSize": 35,
        "lineHeight": 39.97,
      },
      Object {},
    ]
  }
>
  test
</Text>
`;
